kernel/build.rs

```diff
@@ -1,6 +1,27 @@
+use deno_core::extension;
+use std::env;
 use std::path::{Path, PathBuf};
 use std::process::Command;
 
+fn create_deno_snapshot() {
+    extension!(deno_extension, js = ["src/analysis/js/stella.js",]);
+    let out_dir =
+        PathBuf::from(env::var_os("OUT_DIR").expect("OUT_DIR environment variable not found"));
+    let snapshot_path = out_dir.join("DENO_SNAPSHOT.bin");
+
+    let _snapshot = deno_core::snapshot_util::create_snapshot(
+        deno_core::snapshot_util::CreateSnapshotOptions {
+            cargo_manifest_dir: env!("CARGO_MANIFEST_DIR"),
+            snapshot_path,
+            startup_snapshot: None,
+            skip_op_registration: false,
+            extensions: vec![deno_extension::init_ops_and_esm()],
+            compression_cb: None,
+            with_runtime_cb: None,
+        },
+    );
+}
+
```

Comment:
Please put a comment explaining what it does. Especially that it build the snapshot that will be later used by a static variable. We may also put a reference to the cargo book about the environment variable, explaining where to put [references to `OUT_DIR`](https://doc.rust-lang.org/cargo/reference/environment-variables.html)